client/lib/jetpack/wpcom-upsell-controller.tsx

```diff
@@ -5,8 +5,10 @@ import { isJetpackSite } from 'calypso/state/sites/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import type { Context } from './types';
 
-export default function upsellSwitch( UpsellComponent: typeof React.Component ): Function {
-	return ( context: Context, next: Function ) => {
+export default function upsellSwitch(
+	UpsellComponent: typeof React.Component
```

Comment:
You didn't touch this bit, but I believe `typeof` is redundant here:

```suggestion
	UpsellComponent: React.Component
```